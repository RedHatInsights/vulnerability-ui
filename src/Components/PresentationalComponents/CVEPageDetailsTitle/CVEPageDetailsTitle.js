import { TextContent, Title } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React from 'react';
import './CVEPageDetailsTitle.scss';

class CVEPageDetailsTitle extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <React.Fragment>
                <TextContent>
                    <Title size="3xl" className="cve-title">{this.props.title}</Title>
                </TextContent>
                <TextContent>
                    <Title size="md">Publish Date: {this.props.publish_date}</Title>
                </TextContent>
            </React.Fragment>
        );
    }
}

CVEPageDetailsTitle.defaultProps = {
    title: '',
    publish_date: ''
};

CVEPageDetailsTitle.propTypes = {
    title: propTypes.string,
    publish_date: propTypes.string
};

export default CVEPageDetailsTitle;
