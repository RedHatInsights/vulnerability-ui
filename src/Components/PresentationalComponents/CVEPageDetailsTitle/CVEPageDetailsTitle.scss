@import "../../../App";

.cve-title {
    font-weight: bold;
}
