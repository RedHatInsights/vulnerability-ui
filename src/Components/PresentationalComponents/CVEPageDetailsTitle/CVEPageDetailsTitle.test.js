import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CVEPageDetailsTitle from './CVEPageDetailsTitle';

describe('CVEPageDetailsTitle component', () => {
    it('should render without props', () => {
        const wrapper = shallow(<CVEPageDetailsTitle />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
