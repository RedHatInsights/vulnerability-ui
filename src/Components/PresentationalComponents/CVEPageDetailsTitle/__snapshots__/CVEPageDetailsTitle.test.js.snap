// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVEPageDetailsTitle component should render without props 1`] = `
<Fragment>
  <TextContent
    className=""
  >
    <Title
      className="cve-title"
      headingLevel="h1"
      size="3xl"
    />
  </TextContent>
  <TextContent
    className=""
  >
    <Title
      className=""
      headingLevel="h1"
      size="md"
    >
      Publish Date: 
    </Title>
  </TextContent>
</Fragment>
`;
